### PR TITLE
Fix Violations of and Reenable `Naming/RescuedExceptionsVariableName`

### PR DESCRIPTION
**Pull request by Hamms**
Fix applied automaticaly with `bundle exec rubocop --auto-correct-all --only Naming/RescuedExceptionsVariableName`

> This cop makes sure that rescued exceptions variables are named as expected.
>
> The `PreferredName` config option takes a `String`. It represents the required name of the variable. Its default is `e`.

I'm not entirely sure how I feel about this one; on the one hand, consistency is nice, but on the other hand variable names represent an opportunity to communicate meaning and intent. On the other hand, however, it doesn't really look like we're taking all that much advantage of that opportunity, so in practice we're not losing all that much by reenabling this rule.

What do y'all think?

## Links

- https://www.rubydoc.info/gems/rubocop/0.75.0/RuboCop/Cop/Naming/RescuedExceptionsVariableName
- https://docs.rubocop.org/rubocop/cops_naming.html#namingrescuedexceptionsvariablename

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: aws/cloudformation/update_certs [hunk 1]
```diff
@@ -29,7 +29,7 @@ common_name = ARGV[0]
 required_certs = ARGV.dup
 begin
   Acmesmith::Command.start ['current', common_name]
-rescue => _
+rescue => _e
```
Comment (cat5inthecradle) [inline aws/cloudformation/update_certs]: Is assigning the exception to a variable necessary? Compared to a bare `rescue`.

File: bin/cron/export_mysql_database_to_redshift [hunk 2]
```diff
@@ -58,9 +58,9 @@ def main
     RedshiftImport.complete_table_import(Cdo::DMS.redshift_schemas_imported_from_database)
     ChatClient.message 'data', "Completed weekly export from Aurora MySQL database to Redshift."
   end
-rescue StandardError => error
-  ChatClient.message 'data', "Error during export from Aurora MySQL database to Redshift #{error.message}", color: 'red'
-  raise error
+rescue StandardError => e
+  ChatClient.message 'data', "Error during export from Aurora MySQL database to Redshift #{e.message}", color: 'red'
```
Comment (cat5inthecradle) [inline bin/cron/export_mysql_database_to_redshift]:
```suggestion
  ChatClient.message 'data', "Error during export from Aurora MySQL database to Redshift #{e}", color: 'red'
```

`e.message` is identical to `e.to_s`. This is the convention we use elsewhere.

File: bin/cron/restart_high_memory_frontend_services [hunk 3]
```diff
@@ -72,8 +72,8 @@ def main
       ChatClient.message 'infra-production', capture(restart_command, raise_on_non_zero_exit: false)
     end
     ChatClient.message 'infra-production', 'Done finding and restarting front end services with high memory utilization'
-  rescue StandardError => error
-    ChatClient.message 'infra-production', "Error finding/restarting frond end services with high memory utilization - #{error.message}"
+  rescue StandardError => e
+    ChatClient.message 'infra-production', "Error finding/restarting frond end services with high memory utilization - #{e.message}"
```
Comment (cat5inthecradle) [inline bin/cron/restart_high_memory_frontend_services]:
```suggestion
    ChatClient.message 'infra-production', "Error finding/restarting frond end services with high memory utilization - #{e}"
```

File: bin/oneoff/remove_password_for_code_studio_admin.rb [hunk 4]
```diff
@@ -69,6 +69,6 @@
     raise ActiveRecord::Rollback.new, "Intentional rollback" if DRY_RUN
     puts "Admin password updated - #{admin_id}"
   end
-rescue StandardError => error
-  puts "Error, admin password is not updated - #{admin_id} / #{error.message}"
+rescue StandardError => e
+  puts "Error, admin password is not updated - #{admin_id} / #{e.message}"
```
Comment (cat5inthecradle) [inline bin/oneoff/remove_password_for_code_studio_admin.rb]:
```suggestion
  puts "Error, admin password is not updated - #{admin_id} / #{e}"
```

File: bin/oneoff/wipe_data/fnd-929_remove_old_code_studio_account_permissions [hunk 5]
```diff
@@ -121,6 +121,6 @@ USERS_TO_REVOKE.each do |user_id|
 
     puts "User ID successfully revoked - #{user_id}"
   end
-rescue StandardError => error
-  puts "Error revoking User ID - #{user_id} / #{error.message}"
+rescue StandardError => e
+  puts "Error revoking User ID - #{user_id} / #{e.message}"
```
Comment (cat5inthecradle) [inline bin/oneoff/wipe_data/fnd-929_remove_old_code_studio_account_permissions]:
```suggestion
  puts "Error revoking User ID - #{user_id} / #{e}"
```

File: bin/oneoff/wipe_data/remove_unused_levelbuilder_users [hunk 6]
```diff
@@ -92,6 +92,6 @@ USERS_TO_DELETE.each do |user_id|
 
     puts "User ID successfully destroyed - #{user_id}"
   end
-rescue StandardError => error
-  puts "Error destroying User ID - #{user_id} / #{error.message}"
+rescue StandardError => e
+  puts "Error destroying User ID - #{user_id} / #{e.message}"
```
Comment (cat5inthecradle) [inline bin/oneoff/wipe_data/remove_unused_levelbuilder_users]:
```suggestion
  puts "Error destroying User ID - #{user_id} / #{e}"
```

File: bin/oneoff/wipe_data/remove_unused_production_users [hunk 7]
```diff
@@ -60,6 +60,6 @@ USERS_TO_DELETE.each do |user_id|
 
     puts "User ID successfully destroyed - #{user_id}"
   end
-rescue StandardError => error
-  puts "Error destroying User ID - #{user_id} / #{error.message}"
+rescue StandardError => e
+  puts "Error destroying User ID - #{user_id} / #{e.message}"
```
Comment (cat5inthecradle) [inline bin/oneoff/wipe_data/remove_unused_production_users]:
```suggestion
  puts "Error destroying User ID - #{user_id} / #{e}"
```

File: dashboard/app/controllers/omniauth_callbacks_controller.rb [hunk 8]
```diff
@@ -462,15 +462,15 @@ def silent_takeover(oauth_user, auth_hash)
...
+    rescue => e
       error_class = lookup_user.migrated? ?
         'Failed to create AuthenticationOption during silent takeover' :
         'Failed to update User during silent takeover'
       # This should never happen if other logic is working correctly, so notify
       # This can happen if the account being taken over is already invalid
       Honeybadger.notify(
         error_class: error_class,
-        error_message: err.to_s,
+        error_message: e.to_s,
```
Comment (cat5inthecradle) [inline dashboard/app/controllers/omniauth_callbacks_controller.rb]:
```suggestion
        error_message: e.message,
```

Slightly more explicit than `to_s`

File: dashboard/test/models/school_test.rb [hunk 9]
```diff
@@ -30,8 +30,8 @@ class SchoolTest < ActiveSupport::TestCase
 
     begin
       School.merge_from_csv(School.get_seed_filename(true), is_dry_run: true)
-    rescue => error
-      assert_includes error.to_s, 'This was a dry run'
+    rescue => e
+      assert_includes e.to_s, 'This was a dry run'
```
Comment (cat5inthecradle) [inline dashboard/test/models/school_test.rb]:
```suggestion
      assert_includes e.message, 'This was a dry run'
```

File: dashboard/test/models/school_test.rb [hunk 10]
```diff
@@ -56,8 +56,8 @@ class SchoolTest < ActiveSupport::TestCase
 
     begin
       School.merge_from_csv(School.get_seed_filename(true), is_dry_run: true, &parse_row)
-    rescue => error
-      assert_includes error.to_s, 'This was a dry run'
+    rescue => e
+      assert_includes e.to_s, 'This was a dry run'
```
Comment (cat5inthecradle) [inline dashboard/test/models/school_test.rb]:
```suggestion
      assert_includes e.message, 'This was a dry run'
```

File: dashboard/test/test_helper.rb [hunk 11]
```diff
@@ -319,9 +319,9 @@ def assert_same_keys(a, b, a_name = 'A', b_name = 'B', message = "Keys don't mat
   def assert_raises_matching(matcher)
     assert_raises do
       yield
-    rescue => err
-      assert_match matcher, err.to_s
-      raise err
+    rescue => e
+      assert_match matcher, e.to_s
```
Comment (cat5inthecradle) [inline dashboard/test/test_helper.rb]:
```suggestion
      assert_match matcher, e.message
```

File: lib/cdo/aws/dms.rb [hunk 12]
```diff
@@ -181,9 +181,9 @@ def start(max_attempts, delay)
         end
 
         CDO.log.info "DMS Task Completed Successfully: #{@arn}"
-      rescue StandardError => error
-        CDO.log.info "Error executing DMS Replication Task #{@arn} - #{error.message}"
-        raise error
+      rescue StandardError => e
+        CDO.log.info "Error executing DMS Replication Task #{@arn} - #{e.message}"
```
Comment (cat5inthecradle) [inline lib/cdo/aws/dms.rb]:
```suggestion
        CDO.log.info "Error executing DMS Replication Task #{@arn} - #{e}"
```

File: lib/cdo/aws/rds.rb [hunk 13]
```diff
@@ -71,8 +71,8 @@ def self.clone_cluster(
           {db_instance_identifier: clone_instance_id},
           {max_attempts: max_attempts, delay: delay}
         )
-      rescue Aws::Waiters::Errors::WaiterFailed => error
-        CDO.log.info "Error waiting for cluster clone instance to become available. #{error.message}"
+      rescue Aws::Waiters::Errors::WaiterFailed => e
+        CDO.log.info "Error waiting for cluster clone instance to become available. #{e.message}"
```
Comment (cat5inthecradle) [inline lib/cdo/aws/rds.rb]:
```suggestion
        CDO.log.info "Error waiting for cluster clone instance to become available. #{e}"
```

File: lib/cdo/aws/rds.rb [hunk 14]
```diff
@@ -114,8 +114,8 @@ def self.delete_cluster(cluster_id, max_attempts = 20, delay = 60)
             db_cluster_parameter_group_name: existing_cluster.db_cluster_parameter_group
           )
         end
-      rescue Aws::RDS::Errors::DBClusterNotFoundFault => error
-        CDO.log.info "Cluster #{cluster_id} does not exist. #{error.message}.  No need to delete it."
+      rescue Aws::RDS::Errors::DBClusterNotFoundFault => e
+        CDO.log.info "Cluster #{cluster_id} does not exist. #{e.message}.  No need to delete it."
```
Comment (cat5inthecradle) [inline lib/cdo/aws/rds.rb]:
```suggestion
        CDO.log.info "Cluster #{cluster_id} does not exist. #{e}.  No need to delete it."
```

File: lib/cdo/aws/rds.rb [hunk 15]
```diff
@@ -132,9 +132,9 @@ def self.wait_until_db_cluster_deleted(db_cluster_id, max_attempts, delay)
             db_clusters.
             first.
             status
-        rescue Aws::RDS::Errors::DBClusterNotFoundFault => error
+        rescue Aws::RDS::Errors::DBClusterNotFoundFault => e
           cluster_state = 'deleted'
-          CDO.log.info "Database Cluster #{db_cluster_id} has been deleted. #{error.message}"
+          CDO.log.info "Database Cluster #{db_cluster_id} has been deleted. #{e.message}"
```
Comment (cat5inthecradle) [inline lib/cdo/aws/rds.rb]:
```suggestion
          CDO.log.info "Database Cluster #{db_cluster_id} has been deleted. #{e}"
```

File: lib/cdo/git_utils.rb [hunk 16]
```diff
@@ -93,7 +93,7 @@ def self.git_revision_branch(branch)
   def self.get_branch_commit_merges(commit)
     commit_json = JSON.parse(open("https://api.github.com/repos/code-dot-org/code-dot-org/commits/#{commit}").read)
     commit_json['commit']['message'].match(/from code-dot-org\/(.*)\n\n/)[1]
-  rescue => _
+  rescue => _e
```
Comment (cat5inthecradle) [inline lib/cdo/git_utils.rb]: another spot where I'm not sure we need to assign a variable. Does this or another linter require that?

**Review by cat5inthecradle — COMMENTED** [review]
Apologies... this review got bigger than I intended.

The general idea of the suggestions is:

* for brevity, prefer `"#{e}"` inside quoted strings, instead of `e.message`
* for clarity & consistency, prefer `e.message` when it will not automatically cast to a string, instead of vs `e.to_s`

I'm in agreement about the preference toward explicit names. In this case the error blocks are all very tiny, and the variable is used within a couple of lines of it's declaration, so I think it's less of an issue. I slightly prefer `error` or `exception`, to `e`, because I'm allergic to single-letter variable names.

**Review by cat5inthecradle — APPROVED** [review]
In addition to my comments above, I'll also just approve this. Because all of my comments suggest additional enhancement, not pointing out any problems with this change.

File: lib/cdo/git_utils.rb (hunk 16 above)
Comment (Hamms) [reply lib/cdo/git_utils.rb]: I definitely thought there was a linter that did require a variable here, but apparently not!

**Comment by Hamms** [comment]
Great review, thank you!

> I slightly prefer error or exception, to e, because I'm allergic to single-letter variable names.

I agree; there is a `PreferredName` configuration variable we can use to determine this for ourselves, and I think this is a case where the slight grossness of overriding default configuration values would be more than offset by a slightly-more-explicit variable name. I'll open a PR off of this one to explore that

**Pull request merged**